client/client-multi/client-c/CMakeLists.txt

```diff
@@ -178,27 +177,38 @@ set(KAA_SOURCE_FILES
         ${KAA_SRC_FOLDER}/kaa.c
         ${KAA_SRC_FOLDER}/kaa_extension.c
         ${KAA_SRC_FOLDER}/platform-impl/common/kaa_htonll.c
-  )
+)
+
+if(WITH_ENCRYPTION)
+    set(KAA_SOURCE_FILES
+        ${KAA_SOURCE_FILES}
+        ${KAA_SRC_FOLDER}/utilities/kaa_aes_rsa.c
+        ${KAA_SRC_FOLDER}/platform-impl/common/encryption_utils.c)
+endif(WITH_ENCRYPTION)
 
 
 # Includes auto-generated Cmake's scripts.
 include(${CMAKE_CURRENT_LIST_DIR}/listfiles/CMakeGen.cmake)
 
 # Kaa include directories
 set(KAA_INCLUDE_DIRS
-        ${KAA_INCLUDE_PATHS}         # Provided by platform cmake
+        ${KAA_INCLUDE_PATHS}
         ${CMAKE_CURRENT_LIST_DIR}/src
-        ${CMAKE_CURRENT_LIST_DIR}/thirdparty/mbedtls
         ${KAA_SRC_FOLDER}
         ${KAA_THIRDPARTY_INCLUDE_DIR})
 
 # Build unit tests.
 include(${CMAKE_CURRENT_LIST_DIR}/listfiles/UnitTest.cmake)
 
+include_directories(SYSTEM ${KAA_INCLUDE_PATHS})
+
+# This already includes mbedtls
+add_subdirectory(tools/kaa_encryption/rsa_key_gen)
+
 # Build Kaa libraries.
 add_library(kaac ${KAA_SOURCE_FILES})
 target_include_directories(kaac PUBLIC ${KAA_INCLUDE_DIRS})
-target_link_libraries(kaac PRIVATE ${KAA_THIRDPARTY_LIBRARIES})
+target_link_libraries(kaac PRIVATE ${KAA_THIRDPARTY_LIBRARIES} kaa_mbedtls)
```

Comment:
Indeed, the `kaa_mbedtls` is exactly the `mbedtls`, but the `KAA_THIRDPARTY_LIBRARIES` should be altered by platform` CMakeLists.txt`s and not the main `CMakeLists.txt` file.